### PR TITLE
Config refactoring

### PR DESCRIPTION
**Pull request by PymZoR**

### CODE REVIEW

**Comment by gjuchault** [comment]
There is a minor coverage loss in `app.js`. Can you add ignore lines as theses are part of something that can't be tested ?

File: src/lib/log.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default moduleToUse => {
             new winston.transports.Console({
                 prettyPrint: true,
                 colorize   : true,
-                level      : config.level,
+                level      : config.log.level || "debug",
```
Comment (gjuchault) [inline src/lib/log.js]: Singles quotes ?

File: src/app.js [hunk 2]
```diff
@@ -90,12 +90,16 @@ app.start = () => {
...
 
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
         models.loadModels().then(() => {
             log.info('Models loaded');
 
-            server.listen(config.port, () => {
-                log.info('Server is listening on port %d', config.port);
+            server.listen(config.http.port, config.http.hostname, (err) => {
+                if (err) {
```
Comment (gjuchault) [inline src/app.js]: Add istanbul ignore here (port still used and all that stuff)

**Comment by PymZoR** [comment]
Nice profile pic :+1:

**Comment by gjuchault** [comment]
Good to go :rocket:

**Pull request merged**